test/e2e/specs/treeview/00-static-treeview.spec.js

```diff
@@ -37,23 +37,132 @@ exports.tests = function (appName, baseUrl) {
           expect(searchDiv.isDisplayed()).toBeTruthy("Search box input is not visible when page loads initially.");
         });
 
-        it('expand all and collapse all button are visible', function(){
+      });
+
+      describe('actions for the page', function() {
+        beforeAll(function() {
+          browser.wait(EC.elementToBeClickable(filterDropDown), browser.params.defaultTimeout);
+          filterDropDown.click();
+          browser.wait(EC.visibilityOf(element(by.id('number-menu'))), browser.params.defaultTimeout)
+          element(by.id('ui-id-2')).click();
+
+          deriva.waitForElementInverse(element.all(by.id("loadIcon")).get(0));
+
+        });
+
+        it('all node list collapse on `Collapse All` click', function(done) {
+          actionBtn.all(by.css('button')).then(function(buttons) {
+            buttons[1].click().then(function() {
+              expect(element(by.id('jstree')).all(by.css('.jstree-open')).count()).toBe(0);
+              done();
+            });
+          }).catch(function(err) {
```

Comment:
Similar comment as last time, you should avoid nesting promises inside each other. This case will work fine (because of your use of done) but it's better to control the flow of the test case using promise chaining:
```js
actionBtn.all(by.css('button')).then(function(buttons) {
  return buttons[1].click();
}).then(function() {
    expect(element(by.id('jstree')).all(by.css('.jstree-open')).count()).toBe(0);
    done();
}).catch(function(err) {
```

Reply:
Okay, will keep that in mind and use promise chaining. Sorry for that.